core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -189,10 +189,15 @@ Queue<HealthCheckContextGroup> contextGroupChain() {
         return contextGroupChain;
     }
 
-    private List<Endpoint> allHealthyEndpoints() {
-        final List<Endpoint> allHealthyEndpoints = new ArrayList<>();
+    @VisibleForTesting
+    List<Endpoint> allHealthyEndpoints() {
         synchronized (contextGroupChain) {
-            final HealthCheckContextGroup newGroup = contextGroupChain.getLast();
+            final HealthCheckContextGroup newGroup = contextGroupChain.pollLast();
```

Comment:
Q) Shouldn't this be `getLast()`?

If `setEndpoints` is called only once but `updateHealth` continuously calls `allHealthyEndpoints`, would `allHealthyEndpoints()` return empty although it shouldn't?